src/honey_badger/epoch_state.rs

```diff
@@ -370,7 +372,10 @@ where
     /// Checks whether the subset has output, and if it does, sends out our decryption shares.
     fn process_subset(&mut self, cs_step: CsStep<N>) -> Result<Step<C, N>> {
         let mut step = Step::default();
-        let cs_outputs = step.extend_with(cs_step, |cs_msg| {
+        let to_fault = |cs_fault: Fault<N, cs::FaultKind>| {
+            Fault::new(cs_fault.node_id, FaultKind::SubsetFault(cs_fault.kind))
+        };
+        let cs_outputs = step.extend_with(cs_step, to_fault, |cs_msg| {
```

Comment:
Actually, we'll never want to change the node ID when converting fault logs, so `extend_with` should only take a map from a fault _kind_ to fault kind:

```rust
step.extend_with(cs_step, FaultKind::SubsetFault, |cs_msg| { … })
```